From 4dca683c6970bb373b897387ea27895bc063b684 Mon Sep 17 00:00:00 2001
From: "Guo, Jade" <jade.guo@intel.com>
Date: Tue, 28 Nov 2023 17:40:55 +0800
Subject: [PATCH 31/54] Fix size mismatch in snprintf for serialno, add a
 missing bit

Pass the whole serial number to avoid conflict.

Tracked-On: OAM-113638
Signed-off-by: Guo, Jade <jade.guo@intel.com>
---
 libkernelflinger/vars.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/libkernelflinger/vars.c b/libkernelflinger/vars.c
index adcd01f..ae33242 100644
--- a/libkernelflinger/vars.c
+++ b/libkernelflinger/vars.c
@@ -843,7 +843,7 @@ char *get_serial_number(void)
 		goto bad;
 	}
 
-	efi_snprintf((CHAR8*)serialno, SERIALNO_MAX_SIZE, (CHAR8*) "%a", bios_serialno);
+	efi_snprintf((CHAR8*)serialno, SERIALNO_MAX_SIZE + 1, (CHAR8*) "%a", bios_serialno);
 
 	for (pos = serialno; *pos; pos++) {
 		/* Replace foreign characters with zeroes */
-- 
2.25.1

